packages/start/src/client/createServerFn.ts

```diff
@@ -123,30 +117,53 @@ export type ConstrainValidator<TValidator> = unknown extends TValidator
       >
     >
 
-type ServerFnBase<
-  TMethod extends Method = 'GET',
-  TResponse = unknown,
-  TMiddlewares = unknown,
-  TValidator = unknown,
-> = {
-  options: ServerFnBaseOptions<TMethod, TResponse, TMiddlewares, TValidator>
+export interface ServerFnMiddleware<TMethod extends Method, TValidator> {
   middleware: <const TNewMiddlewares>(
     middlewares: Constrain<TNewMiddlewares, ReadonlyArray<AnyMiddleware>>,
-  ) => Pick<
```

Comment:
Basically its slightly faster to build from the bottom up then top down. Creating composable interfaces rather than creating a larger type and `Pick` from it